cmake/tools.cmake

```diff
@@ -15,7 +15,7 @@ execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version OUTPUT_VARIABLE COMPILER
 message (STATUS "Using compiler:\n${COMPILER_SELF_IDENTIFICATION}")
 
 # Require minimum compiler versions
-set (CLANG_MINIMUM_VERSION 12)
+set (CLANG_MINIMUM_VERSION 15)
 set (XCODE_MINIMUM_VERSION 12.0)
 set (APPLE_CLANG_MINIMUM_VERSION 12.0.0)
 set (GCC_MINIMUM_VERSION 11)
```

Comment:
Since we don't support gcc anymore, we can remove this line and the mininum version check below?

Reply:
As you said, GCC support is pretty much dead. There is still lots of GCC code sprinkled throughout the build descriptions (e.g. here). I guess the reason why it was not removed yet is that there was still a little bit of hope that it could be revived even though I personally don't think that resurrecting it will give a lot of benefits. In any case, a removal of GCC code should ideally be done in a separate PR.